test/plugin_functional/test_suites/core_plugins/rendering.ts

```diff
@@ -39,11 +39,10 @@ export default function ({ getService }: PluginFunctionalProviderContext) {
     await appsMenu.clickLink(title);
     return browser.execute(() => {
       if (!('__RENDERING_SESSION__' in window)) {
-        // @ts-expect-error upgrade typescript v4.9.5
         window.__RENDERING_SESSION__ = [];
       }
 
-      window.__RENDERING_SESSION__.push(window.location.pathname);
+      window.__RENDERING_SESSION__!.push(window.location.pathname);
```

Comment:
__question:__ Too bad that TS cannot pick up the right type even right after `in` check :cry:

Reply:
yes... I think it doesn't pick it because it's not an `if...else`. We are populating inside the if, and, apparently, TS doesn't catch that 😢